### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by doncem**
### Change description ###

Client is included in various of projects and since transition to spring boot v2 is favoured the next step to bump to v2.1 is natural. Main concern over here is the compulsory dependency of feign. Major jump from v9 to v10 should be desired approach to any service

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,5 @@
-#Mon Dec 17 16:21:35 GMT 2018
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.2.1-bin.zip
```
Comment (timja) [inline gradle/wrapper/gradle-wrapper.properties]: I think you need some publishing updates with this bump?
see an example of how its done on 5 here:
https://github.com/hmcts/properties-volume-spring-boot-starter/blob/master/build.gradle#L100

Comment (doncem) [reply gradle/wrapper/gradle-wrapper.properties]: skimmed through. hope i did not miss a thing. publish local though did not work comparing to linked example ^. but `Main(...)` worked

**Comment by doncem** [comment]
@nitinprabhu added basic integration test to prove application is booting up and performing as expected

File: src/integrationTest/java/uk/gov/hmcts/reform/authorisation/ServiceAuthorisationApiTest.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+@ContextConfiguration(initializers = IntegrationTestInitializer.class)
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class ServiceAuthorisationApiTest {
+
+    @Autowired
+    private ServiceAuthorisationApi s2sApi;
+
+    @Test
+    public void sa() {
```
Comment (nitinprabhu) [inline src/integrationTest/java/uk/gov/hmcts/reform/authorisation/ServiceAuthorisationApiTest.java]: may be a different method name

Comment (doncem) [reply src/integrationTest/java/uk/gov/hmcts/reform/authorisation/ServiceAuthorisationApiTest.java]: Yikes! I've exposed my classic testing: just a quick keyboard bash to make this test going

Comment (doncem) [reply src/integrationTest/java/uk/gov/hmcts/reform/authorisation/ServiceAuthorisationApiTest.java]: p.s. added step in travis scripts

File: src/integrationTest/java/uk/gov/hmcts/reform/authorisation/TestApplication.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+@SuppressWarnings("HideUtilityClassConstructor") // Spring needs a constructor, its not a utility class
+public class TestApplication {
+
+    public static void main(final String[] args) {
+        SpringApplication.run(TestApplication.class, args);
```
Comment (timja) [inline src/integrationTest/java/uk/gov/hmcts/reform/authorisation/TestApplication.java]: don't think this is needed?
Here's a simple set of tests that use most of the clients:
https://github.com/hmcts/ccd-client/tree/master/src/functionalTest/java/uk/gov/hmcts/reform/ccd/client

Comment (doncem) [reply src/integrationTest/java/uk/gov/hmcts/reform/authorisation/TestApplication.java]: true. had to add couple more annotations though

**Pull request merged**